packages/contracts-bedrock/scripts/deploy/InteropMigration.s.sol

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Script } from "forge-std/Script.sol";
+import { BaseDeployIO } from "scripts/deploy/BaseDeployIO.sol";
+import { IOPContractsManagerInteropMigrator, IOPContractsManager } from "interfaces/L1/IOPContractsManager.sol";
+import { Duration, Proposal, Hash } from "src/dispute/lib/Types.sol";
+import { DeployUtils } from "scripts/libraries/DeployUtils.sol";
+import { IDisputeGameFactory } from "interfaces/dispute/IDisputeGameFactory.sol";
+import { IOptimismPortal2 as IOptimismPortal } from "interfaces/L1/IOptimismPortal2.sol";
+
+contract InteropMigrationInput is BaseDeployIO {
+    address internal _prank;
+    IOPContractsManager internal _opcm;
+
+    bool internal _usePermissionlessGame;
+
+    // starting anchor proposal
+    bytes32 internal _startingAnchorRoot;
+    uint256 internal _startingAnchorL2SequenceNumber;
+
+    // game parameters
+    address internal _proposer;
+    address internal _challenger;
+    uint256 internal _maxGameDepth;
+    uint256 internal _splitDepth;
+    uint256 internal _initBond;
+    uint256 internal _clockExtension;
+    uint256 internal _maxClockDuration;
+
+    bytes internal _opChainConfigs;
+
+    function set(bytes4 _sel, address _value) public {
+        require(address(_value) != address(0), "InteropMigrationInput: cannot set zero address");
+
+        if (_sel == this.prank.selector) _prank = _value;
+        else if (_sel == this.opcm.selector) _opcm = IOPContractsManager(_value);
+        else if (_sel == this.proposer.selector) _proposer = _value;
+        else if (_sel == this.challenger.selector) _challenger = _value;
+        else revert("InteropMigrationInput: unknown selector");
+    }
+
+    function set(bytes4 _sel, bool _value) public {
+        if (_sel == this.usePermissionlessGame.selector) _usePermissionlessGame = _value;
+        else revert("InteropMigrationInput: unknown selector");
+    }
+
+    function set(bytes4 _sel, uint256 _value) public {
+        if (_sel == this.maxGameDepth.selector) {
+            require(_value != 0, "InteropMigrationInput: maxGameDepth cannot be 0");
+            _maxGameDepth = _value;
+        } else if (_sel == this.splitDepth.selector) {
+            require(_value != 0, "InteropMigrationInput: splitDepth cannot be 0");
+            _splitDepth = _value;
+        } else if (_sel == this.initBond.selector) {
+            require(_value != 0, "InteropMigrationInput: initBond cannot be 0");
```

Comment:
Aren't we passing 0 as the init bond in interopgen?

Reply:
Yes. However, when this script is called via interopgen, the `InteropMigrationInput` is bound to the [Go implementation](https://github.com/ethereum-optimism/optimism/blob/0467d3c327610c9740cf4ca236696e697f686310/op-deployer/pkg/deployer/interop/migrate.go#L13) rather than `InteropMigrationInput.initBond` in solidity. The interopgen binding does not restrict the bond amount as this check in solidity doesn't apply there.

The solidity check is used only when you directly run the `InteropMigration.s.sol` script using foundry.
I only kept the interopgen bond amount zero so it's similar to the old system. In any case, the required bond for super DG games will escalate as expected even if the initial bond requirement is zero.

Reply:
Yeah that's a really weird bit of op-deployer.  I suspect we shouldn't write require statements there at all since they are so often skipped and can be misleading. At some point we should set an init bond in tests and have the helpers pass it in so its more production like, but its low priority I think.